frb_codegen/src/generator/rust/ty_enum.rs

```diff
@@ -232,7 +232,16 @@ impl TypeRustGeneratorTrait for TypeEnumRefGenerator<'_> {
                                     self.context.ir_file,
                                     self.context.config,
                                 );
-                                gen.convert_to_dart(field.name.rust_style().to_owned())
+                                let struct_name = field.ty.mirrored_nested();
+                                if field.mirrored_enum && struct_name.is_some() {
+                                    format!(
+                                        "mirror_{}({}).into_dart()",
+                                        struct_name.unwrap(),
+                                        field.name.rust_style().to_owned()
```

Comment:
Disclaimer: Busy with my code so the suggestions may be inperfect!

Hmm if I understand correctly, this fix is like: The `IrField.ty`'s type name is the original struct name, instead of the mirrored struct name, but we want the latter. Thus, here we explicitly hack it to add `mirror_` prefix.

However, this looks like a general problem: What if other places also wants the mirrored name instead of the old name? But the current solution adds `mirrored_enum` flag *inside* IrField.

Ideally, maybe we can have somewhere *general* (instead of tied to IrField) showing whether a type is a mirrored enum. For example, can we ad a field in `IrType`?

Then we can do sth like

```
impl IrType {
  fn the_name_with_possibly_mirror() { ... }
}
```

and use it here happily.

Or, if we are sure every place wants the mirror instead of original type, then just make `the_name_with_possibly_mirror` the default choice, e.g. name it as `name()`

Reply:
Hmm, ok I’ll look into that

Reply:
The issue is solely for enums, so while I get the idea, I don’t think it’s useful to change the entire structure. I added a bunch more tests to make sure

Reply:
what do you think of this way? I added the settings field, cause it feels a bit more *clean* then starting to bloat the IrField

Reply:
Hmm I am still a bit confused.

The current code is: For each *field*, if the *type* of the field is indeed a mirror, do extra log to code generation (addd "mirror_").

However, we know it is *type* that can be a mirror. So what I proposes is: Is it possible we add `is_mirror` or something like that onto the *type*, instead of adding it as a flag for the *field*?

Reply:
Nvm, this is impossible because type is an enum ;)

Reply:
Technically it is possible, but that would be a lot of code, for an edge case. Because then it would have to become a type trait…, and be implemented for all of them. and then again this is only an issue for enums but the code generation would change a lot

Reply:
should I change it, or are we sticking to this approach?

Reply:
> that would be a lot of code, for an edge case

IMHO it should be only a few dozen lines of code.

Reply:
Yes and no, the trait implementation would have to be for all types right? + The current ty approach would have to change to be changed at a later point (that's rather fine)

Reply:
Let me use a few minutes and see whether I can make a patch demonstrating my idea

Reply:
Ofc, I just took another look myself but it would mean that:
1. New function in IrType, that essentially does the same mirrored_nested (easily doable)
 a. Added field in StructRef, that is then used by the function
2. The current logic does more or less sth in this way:
```rust
let gen = TypeRustGenerator::new(
    field.ty.clone(),
    self.context.ir_file,
    self.context.config,
);

gen.convert_to_dart(
    field.try_name_mirror(field.name.rust_style().to_owned()),
)
```
we would have to rewrite the logic both in `ty_enum` and `ty_struct`, and they right now use `field.name` everywhere. Then essentially type needs to know its field or the field function passes itself along to the type? 

IMHO this backlink sounds rather ugly, but maybe I’m missing sth in the code

Reply:
I see the problem: The bug is caused by missing a newtype wrapper, so the current way may be OK, let me check it.